skyportal/models/gwdetector.py

```diff
@@ -0,0 +1,43 @@
+__all__ = ['GWDetector']
+
+import sqlalchemy as sa
+
+from baselayer.app.models import (
+    Base,
+    CustomUserAccessControl,
+    DBSession,
+    public,
+)
+
+from baselayer.app.env import load_env
+
+_, cfg = load_env()
+
+
+def manage_gwdetector_access_logic(cls, user_or_token):
+    if user_or_token.is_system_admin:
+        return DBSession().query(cls)
+    elif 'Manage allocations' in [acl.id for acl in user_or_token.acls]:
+        return DBSession().query(cls)
+    else:
+        # return an empty query
+        return DBSession().query(cls).filter(cls.id == -1)
+
```

Comment:
We are going to need to figure out how to do this without DBSession()

Reply:
yeah this is borrowed from the other handlers. We can fix this in a future PR all at once.

Reply:
Agreed.